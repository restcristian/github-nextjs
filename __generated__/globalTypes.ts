/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

//==============================================================
// START Enums and Input Objects
//==============================================================

/**
 * The possible states of an issue.
 */
export enum IssueState {
  CLOSED = "CLOSED",
  OPEN = "OPEN",
}

//==============================================================
// END Enums and Input Objects
//==============================================================
