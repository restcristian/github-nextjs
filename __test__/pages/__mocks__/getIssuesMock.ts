import { GET_ISSUES_QUERY } from "@/graphql/queries";

export const getIssuesMock = [
  {
    request: {
      query: GET_ISSUES_QUERY,
      variables: {
        owner: "reactjs",
        name: "reactjs.org",
        status: null
      },
    },
    result: {
      data: {
        repository: {
          issues: {
            edges: [
              {
                node: {
                  title: "[Beta] Remove runtime CSS-in-JS",
                  author: {
                    login: "gaearon",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/810438?u=36b11d1db9c2380295b4371a98de87ab6b4c898f&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-25T20:52:44Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "[Beta] DocSearch Title Gets Escaped",
                  author: {
                    login: "Huxpro",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/5563315?u=eae979b4d6884dad781c03ab81e7a16108c7fae1&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-25T21:24:03Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "[Beta] Issue in the navigation of the React Docs Page",
                  author: {
                    login: "fahad0001",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/22472928?u=1d73aa50468118ac9dd2bbc5366eac1620cce0fe&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-27T22:17:07Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Redux and functional components missing in react js",
                  author: {
                    login: "503sayantan",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/79701602?u=0f2fe0968b80a96c425b8a47f053afb1a57c10ce&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-28T07:22:53Z",
                  state: "CLOSED",
                  closedAt: "2022-09-28T07:23:41Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "The issue is i have a component which is having search and select ",
                  author: {
                    login: "gdinesh029",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/3624443?v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-29T00:54:25Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Clock examples use impure `.toLocaleTimeString()`",
                  author: {
                    login: "jonaskuske",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/30421456?u=7dd09d7c52ee9562b6af6dd0174d551f43a50e4d&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-29T11:51:29Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Always exited to hear newer updates ðŸ”¥ React ðŸš€",
                  author: {
                    login: "prathameshbelurkar",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/76102350?u=bd250a5d33fe58dfa1351dc456fe4cbf23677735&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-30T06:39:22Z",
                  state: "CLOSED",
                  closedAt: "2022-10-02T17:46:31Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Switch CDN provider in documentation",
                  author: {
                    login: "ayuhito",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/38220115?u=2ecd63e5c0d379ed144b37a071a6e93cf817f572&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-09-30T14:32:30Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "dependencies",
                  author: {
                    login: "Yunush1",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/105458936?v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-03T08:57:06Z",
                  state: "CLOSED",
                  closedAt: "2022-10-03T08:59:19Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Add popular open source community: PrimeReact ",
                  author: {
                    login: "dcyaner",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/74204161?u=add1c9f122b903e72abad70ff6b1ab2d3b33ab0f&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-03T12:34:45Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Update react testing doc section for react 18",
                  author: {
                    login: "edmundsj",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/11085127?u=c736087a4e4aa261813dbf2d8ffddb7413044f4a&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-04T01:08:38Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "React+Vite",
                  author: {
                    login: "ramiljs",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/89657343?u=e2f2fc309ff392c75bcbe27e301549e2c442998b&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-04T07:55:55Z",
                  state: "CLOSED",
                  closedAt: "2022-10-04T07:57:35Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "Strict mode making newbie like me get confused on debugging.",
                  author: {
                    login: "Misheelajil",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/55972822?v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-05T07:28:41Z",
                  state: "CLOSED",
                  closedAt: "2022-10-06T04:04:17Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "How to avoid re-renders on history.push?",
                  author: {
                    login: "ShrutiJaiswal1494",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/79670853?v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-07T18:34:39Z",
                  state: "CLOSED",
                  closedAt: "2022-10-08T08:52:44Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title: "Update `AUTHORS` file",
                  author: {
                    login: "akulsr0",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/43666833?u=f0bb33f3c5a0eef95542103fdef7bd78282b1f4c&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-08T15:27:09Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "[Beta] Bug: Gutter text overlaps with sandpack code on mobile",
                  author: {
                    login: "shanpriyan",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/42532987?u=e9bb00533d458b0a6740301f5687472be1325887&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-09T13:53:12Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "Beta reactjs 404 & 500 Custom Error pages similarities",
                  author: {
                    login: "vigneshshiv",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/10242660?u=187491df153fc7d6726297d230bbf874fd1daea6&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-09T18:36:16Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    'Clarification about multiple contexts on "Scaling Up with Reducer and Context" documentation page',
                  author: {
                    login: "samanpwbb",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/108094?u=f38ea53150b7855f77de960bde5d8f2cb09aa6ed&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-11T17:56:42Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "pass the set state function to a non-descendent component ",
                  author: {
                    login: "cdhit",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/1721125?u=dfe9da22f98152f9201d022e82c2a8ad98385b68&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-12T01:35:29Z",
                  state: "CLOSED",
                  closedAt: "2022-10-14T06:02:27Z",
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
              {
                node: {
                  title:
                    "Content issue on `reactjs.org/tutorial/tutorial.html`",
                  author: {
                    login: "RitikJaiswal75",
                    avatarUrl:
                      "https://avatars.githubusercontent.com/u/57758447?u=570d8084eb6a5605098c294aa7600602abc7997a&v=4",
                    __typename: "User",
                  },
                  createdAt: "2022-10-15T05:01:09Z",
                  state: "OPEN",
                  closedAt: null,
                  __typename: "Issue",
                },
                __typename: "IssueEdge",
              },
            ],
            __typename: "IssueConnection",
          },
          __typename: "Repository",
        },
      },
    },
  },
];
