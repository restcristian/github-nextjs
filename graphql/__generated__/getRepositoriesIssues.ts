/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: getRepositoriesIssues
// ====================================================

export interface getRepositoriesIssues_repository_issues_edges_node {
  __typename: "Issue";
  /**
   * Identifies the issue title.
   */
  title: string;
}

export interface getRepositoriesIssues_repository_issues_edges {
  __typename: "IssueEdge";
  /**
   * The item at the end of the edge.
   */
  node: getRepositoriesIssues_repository_issues_edges_node | null;
}

export interface getRepositoriesIssues_repository_issues {
  __typename: "IssueConnection";
  /**
   * A list of edges.
   */
  edges: (getRepositoriesIssues_repository_issues_edges | null)[] | null;
}

export interface getRepositoriesIssues_repository {
  __typename: "Repository";
  /**
   * A list of issues that have been opened in the repository.
   */
  issues: getRepositoriesIssues_repository_issues;
}

export interface getRepositoriesIssues {
  /**
   * Lookup a given repository by the owner and repository name.
   */
  repository: getRepositoriesIssues_repository | null;
}

export interface getRepositoriesIssuesVariables {
  owner: string;
  name: string;
}
