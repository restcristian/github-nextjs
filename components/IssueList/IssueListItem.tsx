import React from "react";
import Image from "next/image";
import { getRepositoriesIssues_repository_issues_edges_node } from "@/graphql/__generated__/getRepositoriesIssues";
import styles from "./IssueListItem.module.scss";
import Badge from "../Badge";
import { IssueState } from "@/__generated__/globalTypes";

export interface Props {
  node?: getRepositoriesIssues_repository_issues_edges_node;
}

const IssueListItem: React.FC<Props> = ({ node }) => {
  return (
    <li className={styles.IssueListItem}>
      <div className={styles.Wrapper}>
        <Badge status={node?.state as IssueState} />
        <h3>{node?.title}</h3>
        <div className={styles.AuthorBox}>
          <span>
            Authored By <b>{node?.author?.login}</b> on <b>{node?.createdAt}</b>
          </span>
          <div className={styles.Avatar}>
            <Image
              alt={node?.author?.login}
              data-testid={`avatar-${node?.author?.avatarUrl}`}
              src={{ width: 50, height: 50, src: node?.author?.avatarUrl }}
            />
          </div>
        </div>
      </div>
    </li>
  );
};

export default IssueListItem;
