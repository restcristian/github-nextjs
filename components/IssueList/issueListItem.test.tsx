import { render, screen } from "@testing-library/react";
import { MockedProvider } from "@apollo/client/testing";
import { getIssuesMock } from "@/__test__/pages/__mocks__/getIssuesMock";
import IssueListItem, {Props as IssueListItemProps} from "./IssueListItem";
import { IssueState } from "@/__generated__/globalTypes";

describe("IssueListItem", () => {
  const props: IssueListItemProps = {
      node: {
        title: 'title',
        author: {
            avatarUrl: 'https://avatars.githubusercontent.com/u/9398934?v=4',
            login: 'login_user',
            __typename:"User"
        },
        closedAt: 'closed_date',
        createdAt: '2022-10-03T08:57:06Z',
        state: IssueState.OPEN,
        __typename: "Issue"
      }
  }
  let component: any; 
  beforeEach(() => {
    component = render(
      <MockedProvider mocks={getIssuesMock} addTypename={false}>
        <IssueListItem {...props} />
      </MockedProvider>
    );
  })
  it('should render an issue list item with needed props', async () => {
    expect(await screen.findByText(props?.node?.title as string)).toBeInTheDocument()
    expect(await screen.findByTestId(`avatar-${props?.node?.author?.avatarUrl}` as string)).toBeInTheDocument()
    expect(await screen.findByText(props?.node?.author?.login as string)).toBeInTheDocument()
    expect(await screen.findByText(props?.node?.createdAt as string)).toBeInTheDocument()

    expect(component).toMatchSnapshot();
  });
});
