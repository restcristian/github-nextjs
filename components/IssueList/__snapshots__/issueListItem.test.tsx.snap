// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueListItem should render an issue list item with needed props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="IssueListItem"
      >
        <div
          class="Wrapper"
        >
          <span
            class="Badge"
            data-is-open="true"
            data-testid="badge"
          >
            OPEN
          </span>
          <h3>
            title
          </h3>
          <div
            class="AuthorBox"
          >
            <span>
              Authored By 
              <b>
                login_user
              </b>
               on 
              <b>
                2022-10-03T08:57:06Z
              </b>
            </span>
            <div
              class="Avatar"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="login_user"
                  data-nimg="intrinsic"
                  data-testid="avatar-https://avatars.githubusercontent.com/u/9398934?v=4"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="IssueListItem"
    >
      <div
        class="Wrapper"
      >
        <span
          class="Badge"
          data-is-open="true"
          data-testid="badge"
        >
          OPEN
        </span>
        <h3>
          title
        </h3>
        <div
          class="AuthorBox"
        >
          <span>
            Authored By 
            <b>
              login_user
            </b>
             on 
            <b>
              2022-10-03T08:57:06Z
            </b>
          </span>
          <div
            class="Avatar"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2750%27%20height=%2750%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="login_user"
                data-nimg="intrinsic"
                data-testid="avatar-https://avatars.githubusercontent.com/u/9398934?v=4"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
